pkg/cmd/pulumi/packagecmd/package_info_test.go

```diff
@@ -0,0 +1,194 @@
+				},
+				Enum: []schema.EnumValueSpec{
+					{
+						Name:  "EnumValue1",
+						Value: "value1",
+					},
+					{
+						Value: "value2",
+					},
+				},
+			},
+		},
+	}
+	marshalled, err := json.Marshal(spec)
+	require.NoError(t, err)
+	return marshalled
+}
+
+func TestPackageInfo(t *testing.T) {
+	t.Parallel()
+
+	schema := generateSchema(t)
+	tmpDir := t.TempDir()
+	schemaPath := filepath.Join(tmpDir, "schema.json")
+
+	err := os.WriteFile(schemaPath, schema, 0o600)
+	require.NoError(t, err)
+
+	cmd := newPackageInfoCmd()
+	cmd.SetArgs([]string{schemaPath})
+	var output bytes.Buffer
+	cmd.SetOutput(&output)
+	err = cmd.Execute()
+	require.NoError(t, err)
+	require.Equal(t, fmt.Sprintf(`\x1b[1mName\x1b[0m: test
+\x1b[1mVersion\x1b[0m: 0.0.1
+\x1b[1mDescription\x1b[0m: test description markdown formatted
+\x1b[1mTotal resources\x1b[0m 3
+\x1b[1mTotal modules\x1b[0m: 2
+
+\x1b[1mModules\x1b[0m: another, index
+
+Use 'pulumi package info %[1]s --module <module>' to list resources in a module
+Use 'pulumi package info %[1]s --resource <resource>  --module <module>' for detailed resource info
+`, schemaPath), strings.ReplaceAll(output.String(), "\x1b", "\\x1b"))
+}
+
+func TestModuleInfo(t *testing.T) {
+	t.Parallel()
+
+	schema := generateSchema(t)
+	tmpDir := t.TempDir()
+	schemaPath := filepath.Join(tmpDir, "schema.json")
+
+	err := os.WriteFile(schemaPath, schema, 0o600)
+	require.NoError(t, err)
+
+	cmd := newPackageInfoCmd()
+	cmd.SetArgs([]string{"--module", "index", schemaPath})
+	var output bytes.Buffer
+	cmd.SetOutput(&output)
+	err = cmd.Execute()
+	require.NoError(t, err)
+	require.Equal(t, `\x1b[1mName\x1b[0m: test
+\x1b[1mVersion\x1b[0m: 0.0.1
+\x1b[1mDescription\x1b[0m: test description markdown formatted
+\x1b[1mResources\x1b[0m: 2
+
+ - \x1b[1mTest\x1b[0m: test resource description
+ - \x1b[1mTest2\x1b[0m: this is another test resource
+`, strings.ReplaceAll(output.String(), "\x1b", "\\x1b"))
+}
+
+func TestResourceInfo(t *testing.T) {
+	t.Parallel()
+
+	schema := generateSchema(t)
+	tmpDir := t.TempDir()
+	schemaPath := filepath.Join(tmpDir, "schema.json")
+
+	err := os.WriteFile(schemaPath, schema, 0o600)
+	require.NoError(t, err)
+	cmd := newPackageInfoCmd()
+	cmd.SetArgs([]string{"--module", "index", "--resource", "Test", schemaPath})
+	var output bytes.Buffer
+	cmd.SetOutput(&output)
+	err = cmd.Execute()
+	require.NoError(t, err)
+	require.Equal(t, `\x1b[1mResource\x1b[0m: test:index:Test
+\x1b[1mDescription\x1b[0m: test resource description
+
+\x1b[1mInputs\x1b[0m:
+ - \x1b[1mprop1\x1b[0m (\x1b[4mstring\x1b[0m\x1b[4m*\x1b[0m): this is a string property
+Inputs marked with '*' are required
+
+\x1b[1mOutputs\x1b[0m:
+(All input properties are implicitly available as output properties)
+ - \x1b[1marrayProp\x1b[0m (\x1b[4m[]TestType\x1b[0m\x1b[4m\x1b[0m): this is an array property
+ - \x1b[1menumProp\x1b[0m (\x1b[4menum(string){EnumValue1, value2}\x1b[0m\x1b[4m\x1b[0m): this is an enum property
```

Comment:
This should have object and map items in the test as well, also a ref to a non-enum type

Reply:
Isn't object and map the same?  We have a ref to a non-enum type above, with `[]TestType`.  I've added the object now.

Reply:
objects have Properties, maps have AdditionalProperties